### PR TITLE
10 document the public api

### LINKED ISSUE #10: Document the public API
We need to document the public-facing API of our SDK in order to make it more user-friendly and usable.

At the time of writing this issue, this means that we need to document the methods in the `mistralai_client_dart_base.dart` file. We can use the [official JS client API](https://github.com/mistralai/client-js) as a reference.

### PR DESCRIPTION
**Pull request by MarcinGruchala**
Resolves [issue#10](https://github.com/nomtek/mistralai_client_dart/issues/10)
Changes:
 - Make public API docs more consistent
 - Add docs for embeddings params
 - Add docs for chat params

### CODE REVIEW

File: lib/src/models/chat_params.dart
Comment (gawi151) [inline lib/src/models/chat_params.dart]: I think params docs should be above those params :) At least from what I checked locally.

so for example:
```dart
/// name of the model to chat with, e.g. mistral-tiny
final String model;
```

File: lib/src/models/embeddings.dart
Comment (gawi151) [inline lib/src/models/embeddings.dart]: same as for ChatParams

**Review by gawi151 — COMMENTED** [review]
Suggestion to improve docs

**Review by KacperKluka — DISMISSED** [review]
Looks good but please fix things mentioned by @gawi151 before merging 😉

**Review by gawi151 — APPROVED** [review]
👍

**Comment by gawi151** [comment]
@KacperKluka please take a look again. Me and @MarcinGruchala went through the docs again to make it look better after docs generation.

File: lib/src/models/chat_params.dart
Comment (KacperKluka) [inline lib/src/models/chat_params.dart:52]: I would personally put the `Defaults to XYZ` in separate lines, but that shouldn't block the PR 😉

**Review by KacperKluka — APPROVED** [review]

File: lib/src/models/chat_params.dart
Comment (gawi151) [reply lib/src/models/chat_params.dart:52]: For now, I would keep it as is. If we put it in a separate line then it will be hidden in the generated dart docs (the user will have to go to this field's subpage in docs to see the full description).

**Pull request merged**
